src/main/java/javaslang/control/Kind.java

```diff
@@ -0,0 +1,5 @@
+package javaslang.control;
+
+public interface Kind<TYPE extends Kind<TYPE, ?, ?>, E, T> {
```

Comment:
We call it Kind2 and move out of the `javaslang/control` package into the `javaslang`package.
Maybe I will later generate Kind1..Kindn but that's another story.